### PR TITLE
Enable multi-version and updates binding to v1alpha2

### PR DESCRIPTION
**Pull request by RainbowMango**
**What type of PR is this?**
/kind api-change
/kind feature

**What this PR does / why we need it**:
This PR updates `ResourceBinding` and `ClusterResourceBinding` to `v1alpha2` and servers both `v1alpha1` and `v1alpha2`.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
1. This is a draft PR and is still a work in progress.

**Does this PR introduce a user-facing change?**:

```release-note
The API version of `ResourceBinding` and `ClusterResourceBinding` now has been updated to `v1alpha2`, and the `v1apha1` are still available for one more release.
```

### CODE REVIEW

File: hack/deploy-karmada.sh [hunk 1]
```diff
@@ -103,17 +103,8 @@ function generate_cert_secret {
...
-    kubectl apply -f "${REPO_ROOT}/charts/_crds/bases/policy.karmada.io_overridepolicies.yaml"
-    kubectl apply -f "${REPO_ROOT}/charts/_crds/bases/policy.karmada.io_clusteroverridepolicies.yaml"
-    kubectl apply -f "${REPO_ROOT}/charts/_crds/bases/policy.karmada.io_replicaschedulingpolicies.yaml"
-    kubectl apply -f "${REPO_ROOT}/charts/_crds/bases/work.karmada.io_works.yaml"
-    kubectl apply -f "${REPO_ROOT}/charts/_crds/bases/work.karmada.io_resourcebindings.yaml"
-    kubectl apply -f "${REPO_ROOT}/charts/_crds/bases/work.karmada.io_clusterresourcebindings.yaml"
-    kubectl apply -f "${REPO_ROOT}/charts/_crds/bases/multicluster.x-k8s.io_serviceexports.yaml"
-    kubectl apply -f "${REPO_ROOT}/charts/_crds/bases/multicluster.x-k8s.io_serviceimports.yaml"
+
+    kustomize build "${REPO_ROOT}/charts/_crds" | kubectl apply -f -
```
Comment (lfbear) [inline hack/deploy-karmada.sh:107]: Is it better to use `kubectl kustomize`?  `kustomize` need to install separately

Comment (RainbowMango) [reply hack/deploy-karmada.sh:107]: Wow! I didn't know `kustomize` has integrated into kubectl.
That's great!!!

Comment (lfbear) [reply hack/deploy-karmada.sh:107]: Um, it's integrated since kubectl v1.14

Comment (RainbowMango) [reply hack/deploy-karmada.sh]: Done.

**Pull request merged**